### PR TITLE
feat: test each image before push

### PR DESCRIPTION
**Pull request by KseniaYa**
Tests on images run before the push, if a scenario file for an image is not provided, we skip this step and stop the agent.

### CODE REVIEW

**Comment by damies13** [comment]
This is great progress :+1:  I'll just suggest you might want to add some verification of the test result, check for a specific result name in `_summary.csv` or check for connected in `_agent_data.csv` or open the .db fir (sqlite3) and run a sql query to verify something.

FYI - rfswarm-manager almost always returns 0 (unless there was a python exception that caused a crash)

File: tests/agent/base/scenario.rfs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[Scenario]
+uploadmode = err
+scriptcount = 1
```
Comment (Maresther-B) [inline tests/agent/base/scenario.rfs:3]: If I remember correctly, If you wanted to run scenario[1] and scenario[2], you must set scriptcount=2

File: .github/workflows/check_build_push_template.yml [hunk 2]
```diff
@@ -77,13 +77,31 @@ jobs:
...
+      - name: Create network
+        run: docker network create docker_network
+      - name: Run rfswarm agent
+        run: |
+          docker run --rm -d --name agent --hostname agent --network docker_network \
+          rfswarm-image-for-check
+      - name: Run rfswarm manager or pass
+        run: |
+          if ! [ -f $(pwd)/tests/${{ matrix.image }}/scenario.rfs ]; then
+            echo "Scenario file does not exist, nothing to run."
```
Comment (Maresther-B) [inline .github/workflows/check_build_push_template.yml:92]: OK. So, If I get it right if a scenario file is found in an image directory then , tests are run. I'm I right?

File: .github/workflows/manager_workflow.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 name: Run manager on runner
 on:
-  push:
+  workflow_dispatch:
```
Comment (Maresther-B) [inline .github/workflows/manager_workflow.yml:3]: I didn't get this command

File: tests/agent/seleniumlibrary-firefox/scenario.rfs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+[Scenario]
+uploadmode = err
+scriptcount = 1
```
Comment (Maresther-B) [inline tests/agent/seleniumlibrary-firefox/scenario.rfs:3]: same here, must be incremented

**Review by Maresther-B — COMMENTED** [review]
Goog Job Ksenia. I just suggested the incrementation of scriptcount value to make scenario[1] and [2] running in the test . @damies13, please can you correct me if I'm wrong?

File: tests/agent/base/scenario.rfs (hunk 1 above)
Comment (damies13) [reply tests/agent/base/scenario.rfs:3]: yes correct, if the script count is 1 the manager will only load the first script

File: .github/workflows/check_build_push_template.yml (hunk 2 above)
Comment (damies13) [reply .github/workflows/check_build_push_template.yml:92]: yes, this means when testing base it runs the scenario for base, if it's testing selenium-firefox, it runs that scenario, I think this is a good solution :+1:

File: .github/workflows/manager_workflow.yml (hunk 3 above)
Comment (damies13) [reply .github/workflows/manager_workflow.yml:3]: this just changes how the workflow is triggered in github actions

**Comment by KseniaYa** [comment]
> This is great progress 👍 I'll just suggest you might want to add some verification of the test result, check for a specific result name in `_summary.csv` or check for connected in `_agent_data.csv` or open the .db fir (sqlite3) and run a sql query to verify something.
> 
> FYI - rfswarm-manager almost always returns 0 (unless there was a python exception that caused a crash)

Hello Dave, thank you for your suggestion! I agree that each test result should be checked. For this issue, I just added test runs on CI, but I think we need to work with Esther to prepare the clean controls for the tests.

**Comment by KseniaYa** [comment]
> Goog Job Ksenia. I just suggested the incrementation of scriptcount value to make scenario[1] and [2] running in the test . @damies13, please can you correct me if I'm wrong?

Hi Esther, thanks for this suggestion. I’ve added this modification. If the proposed test launch method ok for you, I’ll merge this request. We can create another issue to update the scenarios and tests to verify the results, as Dave suggests.

**Comment by Maresther-B** [comment]
> > Goog Job Ksenia. I just suggested the incrementation of scriptcount value to make scenario[1] and [2] running in the test . @damies13, please can you correct me if I'm wrong?
> 
> Hi Esther, thanks for this suggestion. I’ve added this modification. If the proposed test launch method ok for you, I’ll merge this request. We can create another issue to update the scenarios and tests to verify the results, as Dave suggests.

Sound good to me

**Pull request merged**